frontend/public/components/markdown-view.tsx

```diff
@@ -18,37 +18,40 @@ const markdownConvert = (markdown, extensions?: string[]) => {
     extensions,
   }).makeHtml(markdown);
 
-  return sanitizeHtml(unsafeHtml, {
-    allowedTags: [
-      'b',
-      'i',
-      'strike',
-      's',
-      'del',
-      'em',
-      'strong',
-      'a',
-      'p',
-      'h1',
-      'h2',
-      'h3',
-      'h4',
-      'ul',
-      'ol',
-      'li',
-      'code',
-      'pre',
-      'button',
+  // add hook to transform anchor tags
+  DOMPurify.addHook("beforeSanitizeElements", function (node) {
+    // nodeType 1 = element type
+    if (node.nodeType === 1 && node.nodeName.toLowerCase() === "a") {
+      node.setAttribute("rel", "noopener noreferrer");
+      return node;
+    }
+  });
+
+  return DOMPurify.sanitize(unsafeHtml, {
+    ALLOWED_TAGS: [
+      "b",
+      "i",
+      "strike",
+      "s",
+      "del",
+      "em",
+      "strong",
+      "a",
+      "p",
+      "h1",
+      "h2",
+      "h3",
+      "h4",
+      "ul",
+      "ol",
+      "li",
+      "code",
+      "pre",
+      "button",
       ...tableTags,
     ],
-    allowedAttributes: {
-      a: ['href', 'target', 'rel', 'data-*'],
-      button: ['class', 'data-*'],
-    },
-    allowedSchemes: ['http', 'https', 'mailto'],
-    transformTags: {
-      a: sanitizeHtml.simpleTransform('a', { rel: 'noopener noreferrer' }, true),
-    },
+    ALLOWED_ATTR: ["href", "target", "rel", "class"],
```

Comment:
Won't this restrict the attributes on all elements, not just `a` elements? I'm wondering if that can break some markup.

Reply:
Hi @spadgett, so from my understanding the previous sanitize-html actually restricted ALL attributes except for 
```
a: ['href', 'target', 'rel', 'data-*'],
button: ['class', 'data-*'],
```
So for example, even though the `code` tag is allowed, the previous config would strip out the attributes.
Markdown:
```
```javascript
alert(s);
```
```

Result HTML before (language class is lost):
```
<code>alert(s);</code>
```

After this change:
```
<code class="javascript language-javascript">alert(s);</code>
```

So technically this is now a little more permissive across all the allowed tags that have been defined.
I didn't see a way to have per-tag rules like sanitize-html had.

Reply:
Looking at which attributes are possible on which elements (https://www.w3schools.com/tags/ref_attributes.asp):
```
href: <a>, <area>, <base>, <link>
target: <a>, <area>, <base>, <form>
rel: <a>, <area>, <form>, <link>
class: all
```
Of these elements the only applicable one based on the currently allowed tags is `<a>`.

So really the only change I can see is that `class` is now allowed on all the tags :)